core/src/main/java/tc/oc/pgm/filters/FilterParser.java

```diff
@@ -453,4 +457,46 @@ public TimeFilter parseTimeFilter(Element el) throws InvalidXMLException {
   public ScoreFilter parseScoreFilter(Element el) throws InvalidXMLException {
     return new ScoreFilter(XMLUtils.parseNumericRange(new Node(el), Integer.class));
   }
+
+  // Methods for parsing QueryModifiers
+
+  @MethodParser("offset")
+  public LocationQueryModifier parseOffsetFilter(Element el) throws InvalidXMLException {
+    String value = el.getAttributeValue("location");
+    // Check vector format
+    XMLUtils.parseVector(new Node(el), value.replaceAll("[\\^~]", ""));
+
+    String[] coords = value.split("\\s*,\\s*");
+
+    Boolean local = null;
+    for (String coord : coords) {
+      if (local == null) {
+        local = coord.startsWith("^");
+      }
+
+      if (coord.startsWith("^") != local)
+        throw new InvalidXMLException("Cannot mix world & local coordinates", el);
+    }
+
+    if (local == null) throw new InvalidXMLException("No coordinates provided", el);
+
+    if (local) {
+      double x = Double.parseDouble(coords[0].substring(1));
+      double y = Double.parseDouble(coords[1].substring(1));
+      double z = Double.parseDouble(coords[2].substring(1));
+      return new LocalLocationQueryModifier(parseChild(el), new Vector(x, y, z));
+    } else {
+      boolean[] relative = new boolean[3];
+      double[] coordinates = new double[3];
+      for (int i = 0; i < coords.length; i++) {
+        String coord = coords[i];
+        boolean isRelative = coord.startsWith("~");
+        relative[i] = isRelative;
+        coordinates[i] = Double.parseDouble(isRelative ? coord.substring(1) : coord);
+      }
+
+      return new WorldLocationQueryModifier(
+          parseChild(el), new Vector(coordinates[0], coordinates[1], coordinates[2]), relative);
```

Comment:
kinda same thing here, no need for coordinates, just use the vector and just check for ~